py/server/deephaven/table_listener.py

```diff
@@ -4,58 +4,243 @@
 
-        Raises:
-            RuntimeError
+        Returns:
+            a generator
         """
+        if not self.j_table_update.removed:
+            return (_ for _ in ())
 
-        if not self.isRegistered:
-            raise RuntimeError("Attempting to deregister an unregistered listener..")
+        col_defs = _col_defs(table=self.table, cols=cols)
+        return _changes_to_numpy(table=self.table, col_defs=col_defs, row_set=self.j_table_update.removed.asRowSet(),
+                                 chunk_size=chunk_size, prev=True)
 
-        self.t.j_table.removeUpdateListener(self.listener)
-        self.isRegistered = False
+    def modified(self, cols: Union[str, List[str]] = None) -> Dict[str, numpy.ndarray]:
+        """Returns a dict with each key bing a column name and each value being a NumPy array of the current values of
+        all the modified rows in the columns.
+
+        Args:
+             cols (Union[str, List[str]): the column(s) for which to return the added rows
+
+        Returns:
+            a dict
+        """
+        if not self.j_table_update.modified:
+            return {}
+
+        col_defs = _col_defs(table=self.table, cols=cols)
+        try:
+            return next(
+                _changes_to_numpy(self.table, col_defs=col_defs, row_set=self.j_table_update.modified.asRowSet(),
+                                  chunk_size=None))
+        except StopIteration:
+            return {}
+
+    def modified_chunks(self, chunk_size: int, cols: Union[str, List[str]] = None) -> Generator[
+        Dict[str, numpy.ndarray], None, None]:
+        """Returns a generator that on each iteration, only returns a chunk of modified rows in the form of a dict with
+        each key bing a column name and each value being a NumPy array of the current values of the rows in the chunk.
+
+        Args:
+            chunk_size (int): the size of the chunk
+            cols (Union[str, List[str]]: the columns(s) for which to return the added rows
+
+        Returns:
+            a generator
+        """
+        if not self.j_table_update.modified:
+            return (_ for _ in ())
+
+        col_defs = _col_defs(table=self.table, cols=cols)
+        return _changes_to_numpy(self.table, col_defs=col_defs, row_set=self.j_table_update.modified.asRowSet(),
+                                 chunk_size=chunk_size)
+
+    def modified_prev(self, cols: Union[str, List[str]] = None) -> Dict[str, numpy.ndarray]:
+        """Returns a dict with each key bing a column name and each value being a NumPy array of the previous values of
+        all the modified rows in the columns.
+
+        Args:
+             cols (Union[str, List[str]): the column(s) for which to return the added rows
+
+        Returns:
+            a dict
+        """
+        if not self.j_table_update.modified:
+            return {}
+
+        col_defs = _col_defs(table=self.table, cols=cols)
```

Comment:
This comment applies everywhere, but I'm just noticing this... should the `_col_defs` function get moved into `_changes_to_numpy`, since that would reduce scope and code complexity?